### DISCUSSION TITLE
How to Initialize Auth With indexedDBLocalPersistence in Nuxt?

### DISCUSSION THREAD

**Discussion by aaronksaunders** [vuejs/vuefire#1349] [Questions and Help]
```
  const auth = initializeAuth(getApp(), {
    persistence: indexedDBLocalPersistence,
  });
```

I am working with the Nuxt Ionic Module running the application on the device and getting errors if I dont initialize auth as above.

Is there some way to pass configuration information into the module to control how auth is initialized?

**Comment by posva** [comment]
From what package does `initializeAuth` comes from?

**Comment by aaronksaunders** [comment]
The auth package.

https://firebase.google.com/docs/auth/web/custom-dependencies

On Mon, Apr 17, 2023 at 5:18 AM Eduardo San Martin Morote <
***@***.***> wrote:

> From what package does initializeAuth comes from?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/vuejs/vuefire/discussions/1349#discussioncomment-5634640>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAEAFGP24IUDPBOS2QYJO5DXBUDGHANCNFSM6AAAAAAXATQKXE>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
-- 

--

Aaron K. Saunders
CEO Clearly Innovative Inc
***@***.***
www.clearlyinnovative.com


This email message and any attachment(s) are for the sole use of the
intended recipient(s) and may contain proprietary and/or confidential
information which may be privileged or otherwise protected from disclosure.
Any unauthorized review, use, disclosure or distribution is prohibited. If
you are not the intended recipient(s), please contact the sender by reply
email and destroy the original message and any copies of the message as
well as any attachment(s) to the original message.

**Reply by posva** [reply]
what package exactly, full path

**Reply by aaronksaunders** [reply]
```
import {initializeAuth, browserLocalPersistence, browserPopupRedirectResolver, browserSessionPersistence, indexedDBLocalPersistence} from "firebase/auth";
import {initializeApp} from "firebase/app";

const app = initializeApp({/** Your app config */});
const auth = initializeAuth(app, {
  persistence: [indexedDBLocalPersistence, browserLocalPersistence, browserSessionPersistence],
  popupRedirectResolver: browserPopupRedirectResolver,
});
```

**"firebase/auth"**

**Reply by rvmourik** [reply]
I encounter the same as above, would be nice to get it to work through Vuefire :-)

**Comment by posva** [comment]
Created #1351 to track
